domains/auth/Auth.API/API/Services/OidcProviders/SignaturGruppen.cs

```diff
@@ -47,6 +51,71 @@ public NextStep CreateAuthorizationUri(AuthState state)
         return authorizationUri;
     }
 
+    public async Task<OidcTokenResponse> FetchToken(string code)
+    {
```

Comment:
As mentioned above this also looks like a long method. Maybe it would benefit to look at the `Extract Method` method :)